### ISSUE TITLE
MissingJavadocModule: new Check

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#21165]
Parent Issue #21113 

**Is your feature request related to a problem? Please describe.**

There is no way to require a Javadoc comment on a module declaration. Every other declaration kind that can carry Javadoc has a presence Check: `MissingJavadocType` for `CLASS_DEF`, `INTERFACE_DEF`, `ENUM_DEF`, `RECORD_DEF` and `ANNOTATION_DEF`, `MissingJavadocMethod` for `METHOD_DEF`, `CTOR_DEF` and others, `MissingJavadocPackage` for `PACKAGE_DEF`, and `JavadocVariable` for `VARIABLE_DEF` and `ENUM_CONSTANT_DEF`. A module declaration has none.

A module without Javadoc is currently not reported by anything:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MissingJavadocPackage"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > mjm/module-info.java <<'EOF'
module com.example.app {
    requires java.base;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml mjm/module-info.java

Starting audit...
Audit done.
```

The equivalent file for a package is reported by `MissingJavadocPackage`:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > package-info.java <<'EOF'
package com.example;
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml package-info.java

Starting audit...
[ERROR] package-info.java:1:1: Missing javadoc for package-info.java file. [MissingJavadocPackage]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe the solution you'd like**

A `MissingJavadocModule` Check that visits `MODULE_DEF` and reports when the declaration has no Javadoc.

Config:

```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="MissingJavadocModule"/>
  </module>
</module>
```

Violation:

```java
1: module com.example.app {
2:     requires java.base;
3: }
```

```
1:1: Missing javadoc for module declaration.
```

No violation:

```java
/**
 * Provides the API for accessing relational databases.
 */
module com.example.app {
    requires java.base;
}
```

**Additional context**

Javadoc on a module is real documentation, not decoration. The [doc comment specification](https://docs.oracle.com/en/java/javase/21/docs/specs/javadoc/doc-comment-spec.html) defines `@provides` and `@uses`, both since JDK 9, which "may only appear in the documentation comment for a module declaration", and the javadoc tool generates a module summary page from this comment.

Implementation note: `MissingJavadocPackageCheck` is the model to follow. Its `hasJavadoc` handles the case where the comment sits either as a previous sibling of the declaration or inside the declaration tree when an annotation intervenes. Module Javadoc has exactly the same two placements, inside `ANNOTATIONS` when the module is annotated and as a direct child of `MODULE_DEF` when it is not.

**Comment by romani** [comment]
>Javadoc on a module is real documentation, not decoration.

Please share example in some real project on such documentation.
Please share javadoc tool output and html rendering on some artificial module.

**Comment by vivek-0509** [comment]
```bash
vivek@Viveks-MacBook-Air jdocdemo % mkdir -p /tmp/jdocdemo/src/com.example.app/com/example/api && cd /tmp/jdocdemo

cat > src/com.example.app/module-info.java <<'EOF'
/**
 * Module level javadoc, this one is real documentation.
 *
 * @provides com.example.api.Service the default implementation
 * @uses com.example.api.Service looked up at runtime
 * @since 1.0
 */
module com.example.app {
    /**
     * Javadoc placed on a directive, does this appear anywhere?
     */
    exports com.example.api;

    uses com.example.api.Service;

    provides com.example.api.Service with com.example.api.Impl;
}
EOF
vivek@Viveks-MacBook-Air jdocdemo % cat > src/com.example.app/com/example/api/Service.java <<'EOF'
package com.example.api;

/** A service that can be discovered at runtime. */
public interface Service {
}
EOF
vivek@Viveks-MacBook-Air jdocdemo % cat > src/com.example.app/com/example/api/Impl.java <<'EOF'
package com.example.api;

/** The default implementation. */
public class Impl implements Service {

    /** Creates an instance. */
    public Impl() {
    }
}
EOF
vivek@Viveks-MacBook-Air jdocdemo % 
javadoc -quiet -d out --module-source-path src --module com.example.app
```

<img width="868" height="711" alt="Image" src="https://github.com/user-attachments/assets/806ce20d-5656-49da-9838-35720d0ca5e6" />

**Comment by romani** [comment]
Do you have link to real code that using this?

**Comment by vivek-0509** [comment]
here it is https://github.com/openjdk/jdk/blob/master/src/java.sql/share/classes/module-info.java
https://github.com/openjdk/jdk/blob/master/src/jdk.compiler/share/classes/module-info.java
